drivers/watchdog/wdt_iwdg_stm32.c

```diff
@@ -137,18 +136,14 @@ static int iwdg_stm32_install_timeout(const struct device *dev,
 		return -EINVAL;
 	}
 
-	tickstart = k_uptime_get_32();
-
+	LL_IWDG_Enable(iwdg); /* enable just before the WriteAccess, else it breaks the sequence  */
 	LL_IWDG_EnableWriteAccess(iwdg);
 
 	LL_IWDG_SetPrescaler(iwdg, prescaler);
 	LL_IWDG_SetReloadCounter(iwdg, reload);
 
-	/* Wait for the update operation completed */
-	while (LL_IWDG_IsReady(iwdg) != 0) {
-		if ((k_uptime_get_32() - tickstart) > IWDG_SR_UPDATE_TIMEOUT) {
-			return -ENODEV;
-		}
+	/* Wait for the update operation completed : expecting IWDG to be ready */
+	while (LL_IWDG_IsReady(iwdg) != 1) {
```

Comment:
Ok, keeping the timeout is possible, but not really useful. The prescaler and counter registers _will_ be modified. After the update, : RVU and PVU update bits of the IWDG status reg bits will retrieve 0, and consequently LL_IWDG_IsReady will become true, no doubt.
 It is just a question of waiting enough (IWDG_SR_UPDATE_TIMEOUT). A timeout is required if bits could remain (be stuck) 1. 

BTW why returning -ENODEV instead of -ETIMEDOUT  ?